pkg/pushsync/pushsync.go

```diff
@@ -432,14 +401,27 @@ func (ps *PushSync) pushToClosest(ctx context.Context, ch swarm.Chunk, origin bo
 				}
 			}
 
-			ps.metrics.TotalSendAttempts.Inc()
+			// since we can reach into the neighborhood of the chunk
+			// act as the multiplexer and push the chunk in parallel to multiple peers
+			if ps.fullNode && swarm.Proximity(peer.Bytes(), ch.Address().Bytes()) >= ps.radiusChecker.StorageRadius() {
+				for ; parallelForwards > 0; parallelForwards-- {
+					retry()
+				}
+			}
 
+			action, err := ps.prepareCredit(ctx, peer, ch, origin)
+			if err != nil {
+				retry()
+				continue
+			}
+
+			ps.metrics.TotalSendAttempts.Inc()
 			inflight++
 
 			go func() {
 				ctxd, cancel := context.WithTimeout(ctx, defaultTTL)
 				defer cancel()
```

Comment:
Would it make sense to move context with timeout creation to `ps.pushChunk` func ?